### PR TITLE
Show docs in the IDE

### PR DESCRIPTION
**Pull request by Duhemm**
Builds on top of #4461.

Rendering could certainly be improved, especially for very long docstrings where one needs to scroll to the type info. Maybe the type info should be displayed above the docstring?

I don't know what's the best way to display it would be, but I'm very open to suggestions.

### CODE REVIEW

File: language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala [hunk 1]
```diff
@@ -338,13 +338,17 @@ class DottyLanguageServer extends LanguageServer
...
 
     if (tpw == NoType) new Hover
     else {
-      val str = tpw.show.toString
-      new Hover(List(JEither.forLeft(str)).asJava, null)
+      import dotty.tools.dotc.core.Comments._
+      val symbol = Interactive.enclosingSourceSymbol(trees, pos)
+      val docComment = ctx.docCtx.flatMap(_.docstring(symbol))
+      val markedString = docMarkedString(docComment, tpw.show.toString)
+      new Hover(List(JEither.forRight(markedString)).asJava, null)
```
Comment (olafurpg) [inline language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala]: What if you return two marked strings? One for the docstring and one for the info. The `textDocument/hover` response can return `MarkedString[]` arrays https://microsoft.github.io/language-server-protocol/specification#textDocument_hover

Comment (Duhemm) [reply language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala]: Thanks for the suggestion! I've changed it to return 2 different `MarkedString`s (one with the doc, the other with the type info). Unfortunately, this doesn't improve much the layout on VSCode: it simply adds an horizontal delimiter between the two hovers, but you still need to scroll.

I've inverted the order of the two `MarkedString`.

**Comment by smarter** [comment]
> Rendering could certainly be improved, especially for very long docstrings where one needs to scroll to the type info. Maybe the type info should be displayed above the docstring?

Yes, open a typescript file and see what happens when you hover over something, we should match this. The signature is above the documentation, the two are separate by a thin horizontal line, the documentation is shown as rendered markdown.

**Comment by Duhemm** [comment]
Just rebased. Do we agree that the formatting of the comments can be improved in a separate PR?

File: language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala [hunk 2]
```diff
@@ -340,13 +340,17 @@ class DottyLanguageServer extends LanguageServer
...
 
     val pos = sourcePosition(driver, uri, params.getPosition)
-    val tp = Interactive.enclosingType(driver.openedTrees(uri), pos)
+    val trees = driver.openedTrees(uri)
+    val tp = Interactive.enclosingType(trees, pos)
     val tpw = tp.widenTermRefExpr
 
     if (tpw == NoType) new Hover
     else {
+      import dotty.tools.dotc.core.Comments._
```
Comment (smarter) [inline language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala]: I would put the import at the top of the file.

**Pull request merged**